src/class/hid/hid_device.c

```diff
@@ -412,4 +412,9 @@ bool hidd_xfer_cb(uint8_t rhport, uint8_t ep_addr, xfer_result_t result, uint32_
   return true;
 }
 
-#endif
+void hidd_sof_isr(uint8_t rhport, uint32_t frame_count)
+{
+  (void) rhport;
+  tud_hid_sof_cb(frame_count);
+}
+#endif
```

Comment:
A function executed directly in a synchron manner isn't a callback. Meaning don't put a "_cb" in an "_isr" because these names refer to different things.

In this function only code should be placed which needs to be executed inside the interrupt service routine right after when the start of frame occurred.

Reply:
This is a callback because it's a configurable function defined by the user, not because of how it is called

Reply:
There are naming conventions for a reason. So people who read a function name have an expectation what the function does. Messing around with that only causes issues.

Just have a look at the other TinyUSB callbacks, like `tud_mount_cb()`. The event for it is invoked (but the function isn't executed) in the ISR. It's later called in the main loop from `tud_task()`. But `audiod_sof_isr()` on the other hand is executed in the ISR. So the "_cb" functions are executed in the main loop and the "_isr" functions within the interrupt service routine.

Reply:
You cant just make up conventions

https://en.m.wikipedia.org/wiki/Callback_(computer_programming)

Reply:
It's the way how it used here in this repro. Feel free to prove me wrong by pointing to a function referred as callback in its name which isn't called indirectly in the main loop. Would be appreciated...